packages/models-library/src/models_library/service_settings.py

```diff
@@ -59,11 +66,145 @@ class Config:
         }
 
 
-class ServiceSettings(BaseModel):
-    __root__: List[ServiceSetting]
+class SimcoreServiceSettings(BaseModel):
+    __root__: List[SimcoreServiceSetting]
 
     def __iter__(self):
         return iter(self.__root__)
 
     def __getitem__(self, item):
         return self.__root__[item]
+
+
+class PathsMapping(BaseModel):
+    inputs_path: Path = Field(
+        ..., description="path where the service expects all the inputs folder"
+    )
+    outputs_path: Path = Field(
+        ..., description="path where the service expects all the outputs folder"
+    )
+    other_paths: List[Path] = Field(
+        [],
+        description="optional list of path which contents need to be saved and restored",
+    )
+
+    @validator("other_paths", always=True)
+    @classmethod
+    def convert_none_to_empty_list(cls, v):
+        return [] if v is None else v
+
+    class Config(_BaseConfig):
+        schema_extra = {
+            "examples": {
+                "outputs_path": "/tmp/outputs",  # nosec
+                "inputs_path": "/tmp/inputs",  # nosec
+            }
+        }
+
+
+ComposeSpecModel = Optional[Dict[str, Any]]
+
+
+class SimcoreService(BaseModel):
+    """Validate all the simcores.services.* labels on a service"""
```

Comment:
All image labels should follow [reverse DNS notation](https://docs.docker.com/config/labels-custom-metadata/#key-format-recommendations) , therefore it should be something like ``io.osparc.simcore.service.settings`` ...

Reply:
I think you are confusing `io.simcore.*` labels with `simcore.services.*` labels. The first is for the frontend (users), the later is for the backend (director services).